### PR TITLE
follow the same logic as in the original Zero_Diameter_Fixer.cpp

### PR DESCRIPTION
**Pull request by asanin-epfl**

### CODE REVIEW

File: neuror/zero_diameter_fixer.py [hunk 1]
```diff
@@ -140,43 +138,35 @@ def fix_in_between(point, stack):
...
         fix_in_between(next_point, stack)
     stack.pop()
 
 
-def connect_average(point1, point2):
+def connect_average(start_point, stop_point, count):
     '''Apply a ramp diameter between the two points
 
     Re-implementation of https://bbpcode.epfl.ch/source/xref/sim/MUK/muk/Zero_Diameter_Fixer.cpp#232
     Contrary to the previous implementation the diameter the ramp is computed in term of
```
Comment (arnaudon) [inline neuror/zero_diameter_fixer.py:157]: is this docstring explaining the original differences between MUK and NeuroR?

Comment (asanin-epfl) [reply neuror/zero_diameter_fixer.py:157]: Seems so. thank you for pointing. I will delete it.

Comment (arnaudon) [reply neuror/zero_diameter_fixer.py:157]: But this means that it was not reproducing MUK on purpose, no? I don't quite follow the algorithm, but is it a big change? It does not look too harmful from the tests, only a slightly different interpolation, right?

Comment (asanin-epfl) [reply neuror/zero_diameter_fixer.py:157]: > it was not reproducing MUK on purpose, no?

Yes.

> I don't quite follow the algorithm, but is it a big change?

It's a major change. The test data didn't have cases that can exhibit such change.

Comment (arnaudon) [reply neuror/zero_diameter_fixer.py]: Ah ok, is there any way to know why this change was made for NeuroR? Maybe there was a good reason to depart from the original implementation?

Comment (asanin-epfl) [reply neuror/zero_diameter_fixer.py]: As the docstring says. This new implementation took into account path lengths. The original implementation bluntly distributed diameter among points. The new distributed them smarter. Segments with longer paths got bigger diameters. The only drawback is that we were not aware of this :D.

I propose to keep both implementations but use the original one by default.

**Review by arnaudon — APPROVED** [review]

File: neuror/zero_diameter_fixer.py (hunk 1 above)
Comment (mgeplf) [reply neuror/zero_diameter_fixer.py]: > I propose to keep both implementations but use the original one by default.

That's a good compromise, thanks.

**Review by mgeplf — APPROVED** [review]

File: neuror/zero_diameter_fixer.py (hunk 1 above)
Comment (asanin-epfl) [reply neuror/zero_diameter_fixer.py]: >> I propose to keep both implementations but use the original one by default.

> That's a good compromise, thanks.

@mgeplf Actually, I've made the opposite. By default the new one (smart with path lengths) is used. If you want to use the legacy (MUK) behavior then you need to pass the flag `legacy=True`. Please confirm/reject whether I should revert.

Comment (mgeplf) [reply neuror/zero_diameter_fixer.py]: sorry, should have been more clear; fine to have default be `legacy=False`

**Comment by asanin-epfl** [comment]
Thank you!

**Pull request merged**